backend/src/common/types/repository.type.ts

```diff
@@ -1,9 +1,9 @@
 type Repository<T = unknown> = {
     find(query: Record<string, T>): Promise<T>;
-    findAll(): Promise<T[]>;
+    findAll(query: Record<string, T>): Promise<T[]>;
     create(payload: unknown): Promise<T>;
-    update(): Promise<T>;
-    delete(): Promise<boolean>;
+    update(query: Record<string, T>, payload: unknown): Promise<T>;
+    delete(query: Record<string, T>): Promise<boolean>;
```

Comment:
Should we replace query to id in update and delete ?

Reply:
In my opinion, query is more clear in parent class repository and service because it is not always user id. We can update and delete items by querying different fields. It depends on certain repository implementation